model/reviews.model.js

```diff
@@ -42,31 +43,38 @@ exports.updateReviewById = async (review_id, { inc_votes } = 0) => {
 exports.selectAllReviews = async ({
   sort_by = "created_at",
   order = "DESC",
+  category,
 }) => {
   const validSortBy = ["created_at", "votes"];
   const validOrderBy = ["DESC", "ASC"];
+  const validCategory = [];
 
   let reviewsQueryStr = `
-    SELECT reviews.*, COUNT(comments.review_id) AS comment_count
-    FROM reviews
-    LEFT JOIN comments
-    ON comments.review_id = reviews.review_id
-    GROUP BY reviews.review_id`;
-
-  if (validSortBy.includes(sort_by)) {
-    reviewsQueryStr += ` ORDER BY reviews.${sort_by}`;
-  } else {
+  SELECT reviews.*, COUNT(comments.review_id) AS comment_count
+  FROM reviews
+  LEFT JOIN comments
+  ON comments.review_id = reviews.review_id`;
+
+  if (!validOrderBy.includes(order) || !validSortBy.includes(sort_by)) {
     return Promise.reject({ status: 400, message: "input is not valid" });
   }
 
-  if (validOrderBy.includes(order)) {
-    reviewsQueryStr += ` ${order}`;
-  } else {
-    return Promise.reject({ status: 400, message: "input is not valid" });
+  if (category) {
+    reviewsQueryStr += ` WHERE reviews.category = $1`;
+    validCategory.push(category);
   }
 
-  const { rows } = await db.query(reviewsQueryStr);
+  reviewsQueryStr += ` 
+    GROUP BY reviews.review_id
+    ORDER BY reviews.${sort_by} ${order}`;
 
+  const { rows } = await db.query(reviewsQueryStr, validCategory);
+  if (!rows.length) {
```

Comment:
I've left a note in your tests for categories that don't exist vs categories that exist but don't have any associated articles. This will need some re-thinking at the same time, as both instances shall return an empty array.

Reply:
Alright, Thanks for the feedback Rose, I will implement this in my work now.